### DISCUSSION TITLE
getServerSideProps calling for multiple endpoints?

### DISCUSSION THREAD

**Discussion by RachelVB** [vercel/next.js#14140] [Help]
Hello, 

I've searched high and low to figure this one out. 

Heres my repo:
https://github.com/RachelVB/sfnnextjs

Im creating an application using NextJS and Strapi headless CMS.
What I'm doing is calling getServerSideProps for my 'newsposts' page from Strapi's API and displaying the data on my home index.js page. 
It works great. 

However now I have another endpoint called 'press-news' I also want to display on my index.js homepage. 

I can't call getServerSideProps again cause it wont let me. 
I've tried using getServerSideProps inside my component but it wont see it, all I ever get is this error: 
<img width="1015" alt="Screen Shot 2020-06-12 at 12 22 17 PM" src="https://user-images.githubusercontent.com/44045336/84539463-4b896480-aca8-11ea-98a4-5efe7f0a30e8.png">

Here is the getServerSideProps code I'm using in my index.js page. 

<img width="1015" alt="Screen Shot 2020-06-12 at 12 30 44 PM" src="https://user-images.githubusercontent.com/44045336/84539605-9b682b80-aca8-11ea-9348-795ac978f2fd.png">

Maybe I'm just missing is completely, but how do I render multiple endpoints on a single page?
After this, my next endpoint will be /jobOpenings.

Thank you everyone in advance.

**Comment by alirezazarei28** [comment]
you should use:
` props.press_news`

**Reply by TofeekAwidat** [reply]
where he should use that>??

**Comment by leerob** [comment]
The code for `getServerSideProps` looks fine – my guess is you're not correctly referencing `press_news` in your React component. Agree with @alirezazarei28.

You're free to call as many API endpoints as you'd like in `getServerSideProps`. You should not that this will block the page from rendering until these requests have finished. If possible, it might make more sense to consider static generation with `getStaticProps`/`getStaticPaths`.
